### ISSUE TITLE
MiniZincIDE nondeterministic crash - "stop" button related

### ISSUE THREAD

**Issue by pafnucy** [MiniZinc/libminizinc#143]
Running in MiniZincIDE 2.1.3 of poetry_idecrash.mzn (code below) leads to nondeterministic crashes. 

Correct behavior: model unsatisfiable
Incorrect behavior:  model shows unsatisfiable message, but "run" button is inactive, "stop" button is active and when clicked, MiniZincIDE closes.

mzn-gecode poetry_idecrash.mzn doesn't crash

Usually, one needs to click 2-4 times on run button before seeing incorrect behavior.

poetry_idecrash.mzn  (based on Basic Modelling for Discrete Optimization on Coursera, workshop_5)
```
int: k=3;
int: l=3;
int: m=3;
set of int: ESSTANZA = 1..k+l;
int: other = 0;
set of int: STANZA = 0..k+l;  % other stanzas are not important

enum KIND = { E, S, O };
array[ESSTANZA] of KIND: kind = [ E | i in 1..k ] ++ [ S | i in 1..l ];

array[ESSTANZA] of int: effect = [10,13,14,9,7,6];
array[int] of int: effect0 = [0] ++ effect;

int: n = k+l+m;
set of int: POSN = 1..n;

array[ESSTANZA] of var POSN:  place; % place of each E and S stanza
array[POSN] of var STANZA: what;     % what is in each posn, 0 = other stanza

constraint what[1] != other /\ 
           forall(i in 1..n-1)
                 (what[i] = other /\ what[i+1] = other 
                  -> forall(j in i..n)(what[j] = other));

constraint forall(i in ESSTANZA, j in POSN)(trace("place[\(i)] = \(j) <-> what[\(j)] = \(i)\n", place[i] = j <-> what[j] = i));
solve maximize sum(p in 1..n-1)(abs(effect0[what[p]] - effect0[what[p+1]]));
```

**Comment by pafnucy** [comment]
Running Ubuntu 16.04

This bug doesn't happen with MiniZincIDE version 2.1.2.

**Comment by guidotack** [comment]
Fixed (was the same problem as #144).
